src/org/apache/commons/httpclient/HttpMethodBase.java

```diff
@@ -1368,6 +1374,81 @@ protected void addCookieRequestHeader(HttpState state, HttpConnection conn)
     }
 
     /**
+     * Put all the cookies in a single header line. 
+     * 
+     * Merge the cookies already present in the request 
+     * with the cookies coming from the state. 
+     * 
+     * @param host the host used with this cookies
+     * @param matcher the {@link CookieSpec matcher} used in this context
+     * @param cookies associated with the {@link HttpState state}
+     * 
+     */
+	private void putAllCookiesInASingleHeader(String host, CookieSpec matcher,
+			Cookie[] cookies) {
+		
+		LOG.trace("enter putAllCookiesInASingleHeader(HttpState state, HttpConnection conn, Cookie[] cookies)" );
+		
+		
+		//use a map to make sure we only have one cookie per name
+		HashMap<String, Cookie> mergedCookies = new HashMap<String, Cookie>();		
+		Header[] cookieLineHeaders = getRequestHeaderGroup().getHeaders(HttpHeader.COOKIE);
+		for (Header cookieLineHeader : cookieLineHeaders) {
+			List<Cookie> cookiesHeader = parseCookieHeader(host, cookieLineHeader.getValue());
+			for (Cookie cookieHeader: cookiesHeader){
+				mergedCookies.put(cookieHeader.getName(),cookieHeader);
+			}			
+			// clean the header
+			getRequestHeaderGroup().removeHeader(cookieLineHeader);
+		}
+		//add the cookies coming from the state 
+		for (Cookie cookie : cookies) {
+			mergedCookies.put(cookie.getName(),cookie);
+		}
+		cookies = mergedCookies.values().toArray(new Cookie[mergedCookies.size()]);
+		String s = matcher.formatCookies(cookies);
+		getRequestHeaderGroup()
+				.addHeader(new Header(HttpHeader.COOKIE, s, true));
+	}
+	/**
+	 * Parse a cookie header in a list of headerElement.
+	 * 
+	 * Unfortunatly the method Header.getElements() is not working properly.
+	 * On a line like Cookie: has-js=1; drupal-toolbar=off it only return the 
+	 * first element has-js=1. So we rework this one. 
+	 * 
+	 * This is a "won't fix" known issue as the project ended.
+	 * https://issues.apache.org/jira/browse/HTTPCLIENT-1356
+	 *  
+	 * @param cookieheader
+	 * @return a list of cookies corresponding to name of the cookie and its value, an empty list if empty or not a cookie line 
+	 */
+	public static List<Cookie> parseCookieHeader(String host, String cookieHeaderValue) {
```

Comment:
ok, it's this method I wanted to test in the unit test and I found it useful enough to be public. I made it default and we'll see.